core/dbt/clients/registry.py

```diff
@@ -33,7 +33,7 @@ def _get(path, registry_base_url=None):
     resp = requests.get(url, timeout=30)
     fire_event(RegistryProgressGETResponse(url=url, resp_code=resp.status_code))
     resp.raise_for_status()
-    if resp is None:
+    if resp.json() is None:
```

Comment:
I think this is the only line I don't totally understand. Can you describe what's going on here?

Reply:
You can get a response that is not `None` but `response.json()` is `None`.  This was originally reported as needed to be retried (why raising `ContentDecodingError` was added) by #4601.  #4577 was reported because the problem was never actually resolved because we were never falling into this `if` statement.  This change will force the retires as expected.

Reply:
gotcha. So what does this `json()` method do?  what does it signify when it returns `None`

Reply:
`resp` is a response object (attributes of content, headers, status_code, etc) and the `json()` method returns the json-encoded  _content_ of the response (or raises an error).  

In some cases it is a completely legitimate response.  You made a valid request but there's no data for what you requested.  We do not expect and empty response.  The content of the response should either not be None or we should get an error (and catch said error with `resp.raise_for_status()` on line 35). From what's been gathered it looks like this is happening when something is wrong with the hub.  It's honestly difficult to replicate.  Catching it and logging the full response will hopefully be helpful too.

Reply:
ahh this makes sense. I might add a comment here about having no json encoded data and that being unexpected.

Reply:
Good point.  Added.